src/map/handler/Map.Gesture.js

```diff
@@ -0,0 +1,105 @@
+import {Map} from '../Map';
+import {Handler} from '../../core/Handler';
+import * as DomEvent from '../../dom/DomEvent';
+import * as Util from '../../core/Util';
+import * as DomUtil from '../../dom/DomUtil';
+/*
+ *import * as Browser from '../../core/Browser';
+ */
+
+/*
+ * L.Handler.TouchZoom is used by L.Map to add pinch zoom on supported mobile browsers.
+ */
+
+// @namespace Map
+// @section Interaction Options
+Map.mergeOptions({
+	// @section Touch interaction options
+	// @option gesture: Boolean|String = *
+	// Whether the map can be zoomed and panned by two finger gesture. If
+	// passed `'center'`, it will zoom to the center of the view regardless of
+	// where the touch events (fingers) were. Enabled for touch-capable web
+	// browsers except for old Androids.
+	gesture: true,
+
+	// @option resetTimeout: Integer
+	// Sets the snap timeout after the zoom gesture ends.
+	resetTimeout: 300,
+
+	// @option zoomMultiplier: Float
+	// Sets the zoom multiplier when doing the zoom gesture.
+	zoomMultiplier: -0.05,
+
+	// @option bounceAtZoomLimits: Boolean = true
+	// Set it to false if you don't want the map to zoom beyond min/max zoom
+	// and then bounce back when pinch-zooming.
+	bounceAtZoomLimits: true
+});
+
+export var Gesture = Handler.extend({
+	addHooks: function () {
+		console.log('adding gesture hook');
+		DomUtil.addClass(this._map._container, 'leaflet-touch-zoom');
+		DomEvent.on(this._map._container, 'wheel', this._onGestureStart, this);
```

Comment:
This risks double-handling mousewheel events. Perhaps choosing between `mousewheel` and `wheel` based on the browser capabilities would be neater?

Reply:
As far as I know, `mousewheel` event is deprecated and `wheel` event is universally supported according to MDN matrix: https://developer.mozilla.org/en-US/docs/Web/API/Element/wheel_event
But tbh, I'm willing to learn. I hacked this PR together, because I grew tired of the lack of support on my work mac when browsing OSM, but I'm not a JS proffesional by trade.

Reply:
There is smart handling for `mousewheel` is our code:
https://github.com/Leaflet/Leaflet/blob/d843c3b88486713827d7e860b58bdba75bfbd5a2/src/dom/DomEvent.js#L100-L101

`mousewheel` is deprecated long time ago so `wheel` is mostly ok.
Except that currently only `mousewheel` will be passive (but that probably should be fixed in relevant part of code).